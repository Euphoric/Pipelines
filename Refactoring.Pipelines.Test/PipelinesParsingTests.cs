using System;
using System.Linq;
using System.Text;
using ApprovalTests;
using Microsoft.CodeAnalysis.CSharp;
using Microsoft.CodeAnalysis.CSharp.Syntax;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Refactoring.Pipelines.Test
{
    [TestClass]
    public class PipelinesParsingTests
    {
        [TestMethod]
        public void Parameters()
        {
            var result = PipelineGenerator.Generate(@"
Sandwich FindBestSandwich(ZipCode zipCode) { /* ... */ }
");
            Approvals.Verify(result);
        }

        [TestMethod]
        public void Invocation()
        {
            var result = PipelineGenerator.Generate(@"
                void _() 
                {
                    var peanutButters = PeanutButterShop.GetAvailable(zipCode); 
                }
");
            Approvals.Verify(result);
        }

        [TestMethod]
        public void Joined()
        {
            var result = PipelineGenerator.Generate(@"
                void _(int value1, int value2)
                {
                    var result = F(value1, value2);
                }
");
            Approvals.Verify(result);
        }

        [TestMethod, Ignore]
        public void MyTestMethod()
        {
            var input = @"
                Sandwich FindBestSandwich(ZipCode zipCode)
                {
                    var peanutButters = PeanutButterShop.GetAvailable(zipCode);
                    var jellies = JellyShop.GetAvailable(zipCode);
                    var bestPeanutButter = peanutButters.BestPeanutButter;
                    var bestJelly = jellies.BestJelly;
                    var bestSandwich = Sandwich.Create(bestPeanutButter, bestJelly);
                    return bestSandwich;
                }
";
            var result = PipelineGenerator.Generate(input);
            Approvals.Verify(result);
        }

        // force this assembly to load
        static readonly Type _ = typeof(Microsoft.CodeAnalysis.CSharp.Formatting.CSharpFormattingOptions);
    }
}
