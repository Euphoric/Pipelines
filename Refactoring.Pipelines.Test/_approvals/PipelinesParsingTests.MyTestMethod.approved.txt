var peanutButtersPipe = zipCodePipe.Process(PeanutButterShop.GetAvailable);
var jelliesPipe = zipCodePipe.Process(JellyShop.GetAvailable);
var bestSandwichPipe = peanutButtersPipe.JoinTo(jelliesPipe).Process((p, j) => Sandwich.Create(p.BestPeanutButter, j.BestJelly))
var resultCollector = bestSandwichPipe.Collect();

